components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandlerTest.java

```diff
@@ -0,0 +1,291 @@
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
+import org.wso2.carbon.identity.core.model.SAMLSSOServiceProviderDO;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth2.TestConstants;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenReqDTO;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.token.OauthTokenIssuer;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.identity.sso.saml.SSOServiceProviderConfigManager;
+import org.wso2.carbon.identity.sso.saml.dto.SAMLSSOAuthnReqDTO;
+import org.wso2.carbon.identity.sso.saml.util.SAMLSSOUtil;
+import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.user.core.tenant.TenantManager;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@PowerMockIgnore({"javax.net.*"})
+
+@PrepareForTest({IdentityUtil.class, IdentityTenantUtil.class, IdentityProviderManager.class, OAuth2Util.class,
+        OAuthServerConfiguration.class, OAuth2AccessTokenReqDTO.class, SSOServiceProviderConfigManager.class})
+
+public class SAML2BearerGrantHandlerTest extends PowerMockTestCase {
+
+    private SAML2BearerGrantHandler saml2BearerGrantHandler;
+
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new PowerMockObjectFactory();
+    }
+
+    @Mock
+    private OauthTokenIssuer oauthIssuer;
+
+    @Mock
+    public RealmService realmService;
+
+    @Mock
+    public TenantManager tenantManager;
+
+    @Mock
+    private IdentityProviderManager identityProviderManager;
+
+    @Mock
+    private IdentityProvider identityProvider;
+
+    @Mock
+    private FederatedAuthenticatorConfig federatedAuthenticatorConfig;
+
+    @Mock
+    private SSOServiceProviderConfigManager ssoServiceProviderConfigManager;
+
+    @Mock
+    private OAuthTokenReqMessageContext tokReqMsgCtx;
+
+    @Mock
+    private OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO;
+
+    @Mock
+    private OAuthServerConfiguration oAuthServerConfiguration;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+
+        when(oAuthServerConfiguration.getIdentityOauthTokenIssuer()).thenReturn(oauthIssuer);
+        SAML2BearerGrantHandler saml2BearerGrantHandler = new SAML2BearerGrantHandler();
```

Comment:
Would be better if we could declare this as a class field and init here. Thereafter you can use refer to the instance from your tests.